tests/unit/helpers/questionnaires.test.js

```diff
@@ -284,54 +330,168 @@ describe('getUserQuestionnaires', () => {
     );
   });
 
-  it('should return an empty array if no questionnaire', async () => {
+  it('should return an empty array if course has slots to plan', async () => {
+    const courseId = new ObjectID();
+    const credentials = { _id: new ObjectID() };
     const course = {
-      _id: new ObjectID(),
+      _id: courseId,
       slots: [{ startDate: new Date('2021-04-20T09:00:00'), endDate: new Date('2021-04-20T11:00:00') }],
+      slotsToPlan: [{ _id: new ObjectID() }],
     };
+
+    findOneCourse.returns(SinonMongoose.stubChainedQueries([course]));
+    nowStub.returns(new Date('2021-04-23T15:00:00'));
+
+    const result = await QuestionnaireHelper.getUserQuestionnaires(courseId, credentials);
+
+    expect(result).toMatchObject([]);
+    sinon.assert.notCalled(findOne);
+    SinonMongoose.calledWithExactly(
+      findOneCourse,
+      [
+        { query: 'findOne', args: [{ _id: courseId }] },
+        { query: 'populate', args: [{ path: 'slots', select: '-__v -createdAt -updatedAt' }] },
+        { query: 'populate', args: [{ path: 'slotsToPlan', select: '_id' }] },
+        { query: 'lean', args: [{ virtuals: true }] },
+      ]
+    );
+  });
+
+  it('should return an empty array if no end of course questionnaire', async () => {
```

Comment:
is started and  no end of course questionnaire

Reply:
is ended plutôt ?

Reply:
bien vu !